model/ServerHandler.php

```diff
@@ -0,0 +1,167 @@
+              /*not_in_db*/ 0,
+              /*dont_write_to_db*/ 1,
+              /*select*/ $this->is_superadmin . ' as _can_edit' ),
+            '_can_delete' => pacol( 0,          0,      1,      'int',  '',              '', 0,
+              /*options*/ '',
+              /*not_in_db*/ 0,
+              /*dont_write_to_db*/ 1,
+              /*select*/ $this->is_superadmin . ' as _can_delete' ),
+        );
+    }
+
+    protected function initMsg() {
+        $this->msg['error_already_exists'] = 'pAdminCreate_server_server_text_error';
+        $this->msg['error_does_not_exist'] = 'server_does_not_exist';
+        $this->msg['confirm_delete'] = 'confirm_delete_server';
+        if ($this->new) {
+            $this->msg['logname'] = 'create_server';
+            $this->msg['store_error'] = 'pAdminCreate_server_result_error';
+            $this->msg['successmessage'] = 'pAdminCreate_server_result_success';
+        } else {
+            $this->msg['logname'] = 'edit_server';
+            $this->msg['store_error'] = 'pAdminEdit_server_result_error';
+            $this->msg['successmessage'] = 'server_updated';
+        }
+        $this->msg['can_create'] = $this->is_superadmin;
+    }
+
+    public function webformConfig() {
+        return array(
+            # $PALANG labels
+            'formtitle_create' => 'pAdminCreate_server_welcome',
+            'formtitle_edit' => 'pAdminEdit_server_welcome',
+            'create_button' => 'pAdminCreate_server_button',
+
+            # various settings
+            'required_role' => 'admin',
+            'listview' => 'list.php?table=server',
+            'early_init' => 0,
+        );
+    }
+
+    protected function storemore() {
+        if ($this->new) {
+            if (!$this->server_postcreation()) {
+                $this->errormsg[] = Config::lang('server_postcreate_failed');
+            }
+        }
+        return true;
+    }
+
+    protected function server_postcreation() {
+        $script=Config::read('server_postcreation_script');
+
+        if (empty($script)) {
+            return true;
+        }   
+
+        if (empty($this->id)) {
+            $this->errormsg[] = 'Empty server parameter in server_postcreation';
+            return false;
+        }   
+
+        $cmdarg1=escapeshellarg($this->id);
+        $command= "$script $cmdarg1";
+        $retval=0;
+        $output=array();
+        $firstline='';
+        $firstline=exec($command,$output,$retval);
+        if (0!=$retval) {
+            error_log("Running $command yielded return value=$retval, first line of output=$firstline");
+            $this->errormsg[] = 'Problems running server postcreation script!';
+            return FALSE;
+        }   
+
+        return TRUE;
+
+    }
+
+    protected function server_postdeletion() {
+        $script=Config::read('server_postdeletion_script');
+
+        if(empty($script)) {
+            return true;
+        }
+
+        if (empty($this->id)) {
+        }
```

Comment:
no reaction if $this->id is empty?

I'd tend to error out.

BTW: Just curious - do you have a real usecase for server postcreate/postdelete scripts, or did you implement the hooks just in case someone needs them?